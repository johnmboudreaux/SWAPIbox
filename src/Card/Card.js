import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ cardData, toggleFavorite } ) => {
  const rowInfoData = {...cardData};
  delete rowInfoData.name;
  delete rowInfoData.type;
  delete rowInfoData.isFavorite;
  delete rowInfoData.id;
  let favText = 'Add Fav';
  if (cardData.isFavorite) {
    favText = 'Remove Fav';
  }
  const displayRows = Object.keys(rowInfoData).map( (rowName, index) => {
    let arrayData = rowInfoData[rowName];
    if (Array.isArray(rowInfoData[rowName])) {
      arrayData = rowInfoData[rowName].map((personPlaceOrThing, index) => {
        return <p key={index}>{personPlaceOrThing} </p>;
      });
    }
    return (
      <tr key={`${cardData.name}-${index}`}>
        <td className='section'>
          {rowName}
        </td>
        <td>
          {arrayData}
        </td>
      </tr>
    );
  });

  const cardStyle = cardData.isFavorite ? 'selected-card card' : 'card';

  return (
    <div className={cardStyle}>
      <h2>{cardData.name}</h2>
      <div className='table-container'>
        <table>
          <tbody>
            {displayRows}
          </tbody>
        </table>
      </div>
      <h5 onClick={()=>{
        toggleFavorite(cardData.id);
      }}>{favText}</h5>
    </div>
  );
};

Card.propTypes = {
  cardData: PropTypes.object,
  toggleFavorite: PropTypes.func
};

export default Card;
