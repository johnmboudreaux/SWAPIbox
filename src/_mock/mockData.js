const mockFilms = {
  "results": [
    {
      "title": "A New Hope",
      "opening_crawl": "It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....",
      "release_date": "1977-05-25"
    }
  ]
};

const mockPeople = {
  "results": [
    {
      "name": "Luke Skywalker",
      "height": "172",
      "mass": "77",
      "hair_color": "blond",
      "skin_color": "fair",
      "eye_color": "blue",
      "birth_year": "19BBY",
      "gender": "male",
      "homeworld": "https://enzymeSucks.com/mockData/api/planets/1/",
      "species": ["https://enzymeSucks.com/mockData/api/species/1/"]
    }
  ]
};

const mockPlanets = {
  "results": [
    {
      "name": "Alderaan",
      "climate": "temperate",
      "gravity": "1 standard",
      "terrain": "grasslands, mountains",
      "surface_water": "40",
      "population": "2000000000",
      "residents": ["https://enzymeSucks.com/mockData/api/people/5/"]
    }
  ]
};

const mockVehicles = {
  "results": [
    {
      "name": "Sand Crawler",
      "model": "Digger Crawler",
      "manufacturer": "Corellia Mining Corporation",
      "cost_in_credits": "150000",
      "length": "36.8",
      "max_atmosphering_speed": "30",
      "crew": "46",
      "passengers": "30",
      "cargo_capacity": "50000",
      "consumables": "2 months",
      "vehicle_class": "wheeled"
    }
  ]
};

const mockSpecies = {
  "name": "Human"
};

const mockPerson = {
  "name": "Luke Skywalker"
};

const mockPlanet = {
  "name": "Tatooine"
};

const mockCardDataObj = {
  Homeworld: "Tatooine",
  Population: "200000",
  Species: "Human",
  id: 1507831320587,
  isFavorite: false,
  name: "Luke Skywalker",
  type: "people"
};

module.exports = {
  mockFilms,
  mockPeople,
  mockPerson,
  mockPlanets,
  mockPlanet,
  mockVehicles,
  mockSpecies,
  mockCardDataObj
};
