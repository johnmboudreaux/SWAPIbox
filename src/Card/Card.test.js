import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import { mockCardDataObj } from '../_mock/mockData';

describe('card', () => {
  it('should render correctly and match the snapshot', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<Card
      key="1"
      cardData={mockCardDataObj}
      toggleFavorite={ mockFunc }/>);

    expect(wrapper).toMatchSnapshot();
  });
});
