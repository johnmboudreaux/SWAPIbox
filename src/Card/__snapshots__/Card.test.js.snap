// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should render correctly and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
        Object {
            "Homeworld": "Tatooine",
            "Population": "200000",
            "Species": "Human",
            "id": 1507831320587,
            "isFavorite": false,
            "name": "Luke Skywalker",
            "type": "people",
          }
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Luke Skywalker
</h2>,
        <div
          className="table-container"
>
          <table>
                    <tbody>
                              <tr>
                                        <td
                                                  className="section"
                                        >
                                                  Homeworld
                                        </td>
                                        <td>
                                                  Tatooine
                                        </td>
                              </tr>
                              <tr>
                                        <td
                                                  className="section"
                                        >
                                                  Population
                                        </td>
                                        <td>
                                                  200000
                                        </td>
                              </tr>
                              <tr>
                                        <td
                                                  className="section"
                                        >
                                                  Species
                                        </td>
                                        <td>
                                                  Human
                                        </td>
                              </tr>
                    </tbody>
          </table>
</div>,
        <h5
          onClick={[Function]}
>
          Add Fav
</h5>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <table>
            <tbody>
                        <tr>
                                    <td
                                                className="section"
                                    >
                                                Homeworld
                                    </td>
                                    <td>
                                                Tatooine
                                    </td>
                        </tr>
                        <tr>
                                    <td
                                                className="section"
                                    >
                                                Population
                                    </td>
                                    <td>
                                                200000
                                    </td>
                        </tr>
                        <tr>
                                    <td
                                                className="section"
                                    >
                                                Species
                                    </td>
                                    <td>
                                                Human
                                    </td>
                        </tr>
            </tbody>
</table>,
          "className": "table-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <tbody>
              <tr>
                            <td
                                          className="section"
                            >
                                          Homeworld
                            </td>
                            <td>
                                          Tatooine
                            </td>
              </tr>
              <tr>
                            <td
                                          className="section"
                            >
                                          Population
                            </td>
                            <td>
                                          200000
                            </td>
              </tr>
              <tr>
                            <td
                                          className="section"
                            >
                                          Species
                            </td>
                            <td>
                                          Human
                            </td>
              </tr>
</tbody>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td
                                    className="section"
                  >
                                    Homeworld
                  </td>
                  <td>
                                    Tatooine
                  </td>
</tr>,
                <tr>
                  <td
                                    className="section"
                  >
                                    Population
                  </td>
                  <td>
                                    200000
                  </td>
</tr>,
                <tr>
                  <td
                                    className="section"
                  >
                                    Species
                  </td>
                  <td>
                                    Human
                  </td>
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Luke Skywalker-0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td
                      className="section"
>
                      Homeworld
</td>,
                    <td>
                      Tatooine
</td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Homeworld",
                      "className": "section",
                    },
                    "ref": null,
                    "rendered": "Homeworld",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Tatooine",
                    },
                    "ref": null,
                    "rendered": "Tatooine",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "Luke Skywalker-1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td
                      className="section"
>
                      Population
</td>,
                    <td>
                      200000
</td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Population",
                      "className": "section",
                    },
                    "ref": null,
                    "rendered": "Population",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "200000",
                    },
                    "ref": null,
                    "rendered": "200000",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "Luke Skywalker-2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td
                      className="section"
>
                      Species
</td>,
                    <td>
                      Human
</td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Species",
                      "className": "section",
                    },
                    "ref": null,
                    "rendered": "Species",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Human",
                    },
                    "ref": null,
                    "rendered": "Human",
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
          "type": "table",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Add Fav",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Fav",
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Luke Skywalker
</h2>,
          <div
            className="table-container"
>
            <table>
                        <tbody>
                                    <tr>
                                                <td
                                                            className="section"
                                                >
                                                            Homeworld
                                                </td>
                                                <td>
                                                            Tatooine
                                                </td>
                                    </tr>
                                    <tr>
                                                <td
                                                            className="section"
                                                >
                                                            Population
                                                </td>
                                                <td>
                                                            200000
                                                </td>
                                    </tr>
                                    <tr>
                                                <td
                                                            className="section"
                                                >
                                                            Species
                                                </td>
                                                <td>
                                                            Human
                                                </td>
                                    </tr>
                        </tbody>
            </table>
</div>,
          <h5
            onClick={[Function]}
>
            Add Fav
</h5>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <table>
              <tbody>
                            <tr>
                                          <td
                                                        className="section"
                                          >
                                                        Homeworld
                                          </td>
                                          <td>
                                                        Tatooine
                                          </td>
                            </tr>
                            <tr>
                                          <td
                                                        className="section"
                                          >
                                                        Population
                                          </td>
                                          <td>
                                                        200000
                                          </td>
                            </tr>
                            <tr>
                                          <td
                                                        className="section"
                                          >
                                                        Species
                                          </td>
                                          <td>
                                                        Human
                                          </td>
                            </tr>
              </tbody>
</table>,
            "className": "table-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <tbody>
                <tr>
                                <td
                                                className="section"
                                >
                                                Homeworld
                                </td>
                                <td>
                                                Tatooine
                                </td>
                </tr>
                <tr>
                                <td
                                                className="section"
                                >
                                                Population
                                </td>
                                <td>
                                                200000
                                </td>
                </tr>
                <tr>
                                <td
                                                className="section"
                                >
                                                Species
                                </td>
                                <td>
                                                Human
                                </td>
                </tr>
</tbody>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td
                                        className="section"
                    >
                                        Homeworld
                    </td>
                    <td>
                                        Tatooine
                    </td>
</tr>,
                  <tr>
                    <td
                                        className="section"
                    >
                                        Population
                    </td>
                    <td>
                                        200000
                    </td>
</tr>,
                  <tr>
                    <td
                                        className="section"
                    >
                                        Species
                    </td>
                    <td>
                                        Human
                    </td>
</tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Luke Skywalker-0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        className="section"
>
                        Homeworld
</td>,
                      <td>
                        Tatooine
</td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Homeworld",
                        "className": "section",
                      },
                      "ref": null,
                      "rendered": "Homeworld",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Tatooine",
                      },
                      "ref": null,
                      "rendered": "Tatooine",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": "Luke Skywalker-1",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        className="section"
>
                        Population
</td>,
                      <td>
                        200000
</td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Population",
                        "className": "section",
                      },
                      "ref": null,
                      "rendered": "Population",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "200000",
                      },
                      "ref": null,
                      "rendered": "200000",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": "Luke Skywalker-2",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        className="section"
>
                        Species
</td>,
                      <td>
                        Human
</td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Species",
                        "className": "section",
                      },
                      "ref": null,
                      "rendered": "Species",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Human",
                      },
                      "ref": null,
                      "rendered": "Human",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
            "type": "table",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Add Fav",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Fav",
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
